In [241]:
import nltk
nltk.download('vader_lexicon')

[nltk_data] Downloading package vader_lexicon to C:\Users\Kaden's
[nltk_data]     Laptop\AppData\Roaming\nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


True

In [242]:
!pip install translate-api
!pip install translators --upgrade

Defaulting to user installation because normal site-packages is not writeable
Defaulting to user installation because normal site-packages is not writeable


In [243]:
import translators as ts

In [244]:
import string
%matplotlib inline
import pandas as pd
from nltk import sentiment
from IPython.display import display, Markdown
from nltk.sentiment.vader import SentimentIntensityAnalyzer
from nltk import tokenize
from typing import *

# Sentiment_Analyzer

We begin by importing the data file and looking at a portion of our data

In [245]:
data = pd.read_csv(r"C:\Users\Kaden's Laptop\Documents\GitHub\KadenFranklin.github.io\sentiment_analyzer\names1.csv")
print(data.head())
data.describe()

  Local Names                   Translation    Location Real Names
0      Abombo                             _  Kisanagani     Dupont
1       Adami                         Madam      Opala     generic
2     Adamisu  Knowing eyes, control, anger      Basoko          _
3       Aduyi                          Lion           _          _
4       Aginu                      Kneeling   Kisangani      Genot


Local Names Translation Location Real Names
count          271         271      267        271
unique         270         180       81         99
top      Tuku-Tuku           _  Katanga          _
freq             2          68       28        130

The data gathered for this analyis is collected from King Leopold's Congo and the "Scramble for Africa" by Michael A. Rutz

Specifically from page 163 and the 'Sample of names used in this work'.

We can see from the commands above that there are 271 rows of names, with 270 unique names, 188 distinct translations, originating from 91 locations. In addition to this, asside from collective 'meta' names there are 99 names given to colonials by natives.

The datatypes are as follows:

In [246]:
data.dtypes

Local Names    object
Translation    object
Location       object
Real Names     object
dtype: object

Local Name - Object

Translation - Object

Location - Object	

Real Name - Object

Now just to be certain we will check the dataset for duplicates.

In [247]:
i = -1
for x in data.duplicated():
    i += 1
    
    if x == True:
        print(x)
        print(i)

Success, no duplicates, Now we move on to the interesting part, [sentiment analysis](https://monkeylearn.com/sentiment-analysis/).

Now we can split our data into its corresponding columns so that we can work with it more easily.


In [248]:
names_1 = list(data['Local Names'])
transl = list(data["Translation"])
location = list(data["Location"])
names_2 = list(data['Real Names'])

In [249]:
new_translations = []
for x, y in enumerate(names_1):
    new_translations.append(ts.google(y))

The function below defines the [markdown](https://www.markdownguide.org/) table, which the results of our analysis will be stored.

As you can see there are also some helper function I've created to help iterate through data.

In [250]:
def show_markdown_table(headers: List[str], data: List) -> str:
    s = f"| {' | '.join(headers)} |\n| {' | '.join([(max(1, len(header) - 1)) * '-' + ':' for header in headers])} |\n"
    for row in data:
        s += f"| {' | '.join([str(item) for item in row])} |\n"
    display(Markdown(s))

The function below takes in an input of either one name, or a list of names. It will return a resulting markdown table giving the scores of such input names.

In [251]:
def en_name_analyzer(name):

    sid = SentimentIntensityAnalyzer()    

    return sid.polarity_scores(name)

In [252]:
def sw_name_analyzer(name):
    
    name = ts.google(name)

    sid = SentimentIntensityAnalyzer()    

    return sid.polarity_scores(name)

In [253]:
    #dis=[]
    #sid = SentimentIntensityAnalyzer()
    
    #    for x, elem in enumerate(name):
    #         print(ts.google(elem))

    
    #if isinstance(name, list):
    #    dis.append([[sid.polarity_scores(x)] for x in name])
    #    return dis
                
    #else:
    #    return sid.polarity_scores(name)

Having defined such functions we can perform an analysis on our own dataset.

The code below is complex, but I will annotate it to the best of my ability.
 
 comments are denoted with '#' marks.

In [254]:
headers = ['Local Name', 'Score*', 'Location', 'Translation', 'Real Name']
# This line denotes the first line of output data, our heading

rows = [[names_1[num], str(str(sw_name_analyzer(names)['compound']) + " -" + new_translations[num]), location[num], 
         str( str(en_name_analyzer(trnl)['compound']) + " -" + transl[num]), 
         str( str(en_name_analyzer(name_2)['compound']) + " -" + names_2[num])] 
        for num, (names, name_2, trnl) in enumerate(zip(names_1, names_2, transl))]

# This section of code defines the rows that will be generated.
# It relies on various other functions and lists define above, but this is the most critical section.

# 1st line - local names, their computed translations w/ scores, and location lists
# 2nd line - Translation list
# 3rd line - Real names / origin list
# last line - for #, (variables) in enumerate(three different lists) [This line iterates over all the other to construct the table.]


show_markdown_table(headers, rows)
# This function combines the two lines above to compile output

| Local Name | Score* | Location | Translation | Real Name |
| ---------: | -----: | -------: | ----------: | --------: |
| Abombo | 0.0 -Rb | Kisanagani | 0.0 -_ | 0.0 -Dupont |
| Adami | 0.0 -Adami | Opala  | 0.0 -Madam | 0.0 -generic |
| Adamisu | -0.1531 -Yasuda Miss | Basoko | -0.5719 -Knowing eyes, control, anger | 0.0 -_ |
| Aduyi | 0.0 -Coat | _ | 0.0 -Lion | 0.0 -_ |
| Aginu | 0.0 -Agus | Kisangani | 0.0 -Kneeling | 0.0 -Genot |
| Agomiriya | 0.0 -Will wait | Moto | -0.5574 -Abuser of women; Unfaithful husband | 0.0 -generic |
| Alali na se | 0.0 -He sleeps below | Equateur | -0.5267 -He who is lying down | 0.0 -_ |
| Alinga Mama | 0.3612 -She is like a mother | Aruwimi  | 0.5719 -He who loves mother;  | 0.0 -generic |
| Alinga Likaya | 0.5719 -He loves tobacco | _ | 0.0 -Drunken smoker | 0.0 -_ |
| Alinga Tumba | 0.3612 -He is like a tun | _ | 0.296 -Fight Lover  | 0.0 -_ |
| Alube | 0.0 -Has not been | Kasai | 0.0 -King Albert | 0.0 -King Albert |
| Angbalima | 0.0 -Angbalima | Basoko | -0.5267 -Terror | 0.0 -_ |
| Angbotalikume | 0.0 -Angbotalicume | Baskoko | 0.0 -_ | 0.0 -_ |
| Angwadima | 0.0 -Ugmin | Eastern Congo | -0.5267 -Terror | 0.0 -_ |
| Asali Monkanda | 0.0 -He made a letter | Kisangani | 0.0 -He has written a letter  | 0.0 -D'Hondt |
| Atama-Atama | 0.0 -Assignment-atma | Opala | 0.0 -Enslavers | 0.0 -any whites |
| Atamu | 0.0 -Adam | Opala | 0.0 -Madam | 0.0 -generic |
| Ataso | 0.0 -Atta | everywhere | 0.0 -Attention | 0.0 -generic |
| Atila Okondo | 0.0 -Attila Okondo | Opala | 0.0 -He who does not let any land lie fallow | 0.0 -any crop supervisor |
| Awaya-Awaya | 0.0 -Ashaya | Kasai  | -0.3182 -Lack of direction  | 0.0 -generic |
| Bajunu | 0.0 -Bajun | Kisangani  | 0.0 -Kneeling | 0.0 -generic |
| Bakola | 0.0 -They work | Inongo  | 0.0 -Generation | 0.0 -Cloetens et al. |
| Bala-Bala | 0.0 -Bala-Bala | everywhere any | 0.0 -Main road; Officials on the move | 0.0 -itinerate agent |
| Balikpe | 0.0 -Balikpe | Uele  | 0.0 -Leaves eater  | 0.0 -G. Schweinfurth |
| Balikuhe | 0.0 -Bali Kuhehe | Uele  | 0.0 -He was eating raw, bland food  | 0.0 -G. Miani |
| Bambenga | -0.25 -They are stuck | Equateur | 0.0 -Red pepper  | 0.0 -_ |
| Bana Muanama | 0.0 -Bani Muanama | Kisangani  | 0.1779 -Water spirit  | 0.0 -Wouters |
| Bandungu | 0.0 -Bandungu | Lac Leopold II  | 0.0 -Pepper | 0.0 -P.-J. Van de Moere |
| Bapenda Kula | 0.0 -Bapenda Kula | Ituri  | 0.6369 -They who love to eat  | 0.0 -generic |
| Basikoti | 0.0 -Basic | everywhere | 0.0 -Much whipping; Hippopotamus-hide whip | 0.0 -generic |
| Batu Pamba | 0.0 -The Buet on the | Uele  | 0.0 -Straw men | 0.0 -generic |
| Belei | 0.0 -Confess | Bumba  | 0.5859 -Lover of women  | 0.0 -generic |
| Bikengakenga | 0.0 -Bikely | Katanga | 0.0 -_ | 0.0 -_ |
| Bola-Bola | 0.0 -Was-colored | Equateur | 0.0 -Beat, bear; ruthless beater | 0.0 -_ |
| Bolemba | 0.0 -Bankbar | Équateur, Opala | -0.6597 -Stupefying vine for killing fish | 0.0 -_ |
| Bombeki | 0.0 -Bombeki | Equateur | 0.0 -_ | 0.0 -H. Bombeeck |
| Bosolo | 0.0 -Bosolo | Equateur | 0.0 -_ | 0.0 -_ |
| Bula Matari | 0.0 -Right castle | _ | 0.0 -Break rocks | 0.0 -any agent |
| Bula Matende | 0.0 -Bula therapy | Vivi, Basoko | -0.296 -Dynamiter; Break rocks | 0.0 -Van Kerkhoven |
| Bumba | 0.0 -Ball | Kisangani | 0.0 -Who comes from bumba | 0.0 -Luthy |
| Bwana Kapia | 0.0 -Bwana Kiapia | Katanga  | 0.0 -Second-born of twins  | 0.0 -Delvaux |
| Bwana Kaseya | 0.0 -Bwana Kaseya | Kasongo  | 0.0 -_ | 0.0 -Verditch |
| Bwana Kenengene | 0.0 -Sir | Katanga | 0.0 -_ | 0.0 -_ |
| Bwana Kidogo | 0.0 -Lord a little | Eastern Congo | 0.0 -Mister small; Little boss  | 0.0 -_ |
| Bwana Kioseni | 0.0 -Mr. Kioseni | _ | 0.0 -_ | 0.0 -_ |
| Bwana Kitoko | 0.0 -Bwana Kaitoko | everywhere  | 0.4939 -Mister handsome  | 0.0 -King Baudouin |
| Bwana Leke | 0.0 -Bwana Leke | Kasai | 0.0 -Mister Leke | 0.0 -Legat |
| Bwana Lubuku | 0.0 -Mr. Lubuku | Kasongo | 0.0 -Master builder;  | 0.0 -Van Malderen |
| Bwana Lutshina | 0.0 -Bybya jump | Katanga | 0.0 -_ | 0.0 -_ |
| Bwana Maibwe | 0.0 -Sossa | Katanga  | 0.0 -Mister stone | 0.0 -_ |
| Bwana Mukubwa | 0.0 -Lord | Eastern Congo | 0.5423 -Big boss; Superior; Mister tall  | 0.0 -_ |
| Bwana Mzuri | 0.4404 -Good sir | Katanga | 0.4939 -Mister handsome  | 0.0 -_ |
| Bwana Ndeke  | 0.0 -Boss Down | Kilo | 0.0 -Mister bird  | 0.0 -Henry |
| Bwana Ndeke | 0.0 -Boss Down | Kisangani | 0.0 -Mister bird  | 0.0 -L. Oswald, L. Adhémar, C. de Leuze |
| Bwana Nioka  | 0.0 -Bwana Snake | Katanga | 0.0 -Mister snake | 0.0 -Gilson |
| Bwana Nzige | 0.0 -Lord locust | _ | 0.0 -Mister locust  | -0.5994 -any violent agent |
| Bwana Pangabisoto  | 0.0 -Making Make | Katanga | 0.0 -_ | 0.0 -_ |
| Bwana Tomasi | 0.0 -Mr Thomas | Katanga | 0.0 -Mister Thomas  | 0.0 -_ |
| Bwana Tumbaku | 0.0 -Lord Tobacco | Kisangani | 0.0 -_ | 0.0 -T.Nicolas |
| Chakundia | 0.0 -Chakandia | Katanga | 0.0 -He who can eat me  | 0.0 -_ |
| Chicotte | 0.0 -CHICOTTE | Baringa | 0.0 -Whipping  | 0.0 -G. Dineur |
| Cibalabala | 0.0 -Cibalabala | Kasai  | 0.0 -Wild cat | 0.0 -F. Van Nimmen, J. Van Uden |
| Cibutama | 0.0 -Ci is not fucking | Kasai | -0.1779 -To hide small scar  | 0.0 -R. Bearts |
| Cimpanga | 0.0 -Cimpanga | Kasai  | 0.0 -Bull | 0.0 -A. Demunster |
| Ciswa-bantu | 0.0 -Cishes-help | Kasai  | 0.5719 -He who loves people | 0.0 -A. Lippens |
| Citoko | 0.0 -Citoko | Kasai | 0.4939 -Handsome  | 0.0 -_ |
| Djeke | 0.0 -Djoy | Katanga | 0.0 -_ | 0.0 -_ |
| Djoko deli | 0.0 -Djoko Deli | Kasai | 0.0 -_+ | 0.0 -_ |
| Ebeniq | 0.0 -Benia | Katanga | 0.0 -_ | 0.0 -_ |
| Ebuka-Buka | 0.0 -Looking-out | Ekonda, Équateur | 0.0 -Breaker of things, into pieces | 0.0 -_ |
| Efanja | 0.0 -Font | Ekonda, Équateur | 0.0 -He who disperses people  | 0.0 -_ |
| Ekanda | 0.0 -As you're | Bolobo | 0.0 -_ | 0.0 -Grenfell |
| Ekanga-kanga | 0.0 -Cursten | Équateur | -0.4404 -He who arrests people arbitrarily AND indiscriminately | 0.0 -_ |
| Ekatankoi | 0.0 -Hens | Tomba | 0.0 -_ | 0.0 -Steinbeck |
| Ekuma | 0.0 -To grow | nan | -0.5574 -He who destroys things  | 0.0 -Équateur |
| Eminimbi | 0.3182 -I'm sure | Uele  | -0.5267 -Emin, the wicked | 0.0 -E. Schnitzer (Emin Pacha) |
| Engalala | 0.0 -Blouse | Lac Leopold II | 0.0 -Palm leaf | 0.0 -Lemaire |
| Etumba Mbilo | -0.3818 -Fight Mbilo | nan | -0.3818 -Office that resemblesa battlefield | 0.0 -_ |
| Ewa-Olefe | 0.0 -Ewa-Olefe | Yanonge  | 0.0 -Womanizer  | 0.0 -generic |
| Fasa ou Nkosa Nkosa | 0.0 -Fasa th Gosa Gosa | Inongo | 0.0 -_ | 0.0 -_ |
| Fazzari | 0.0 -Fazzari | Basoko, Mbesa | 0.0 -_ | 0.0 -_ |
| Fimbo Mingi | 0.0 -A lot of rods | Kasongo | 0.0 -Much whipping; Hippopotamus-hide whip | 0.0 -Baron Dhanis |
| Genda genda | 0.0 -Genda Genda | _ | 0.0 -_ | 0.0 -_ |
| Guruguru | 0.0 -round and round | Uele | 0.0 -Pierced ear lobes  | 0.0 -Ibrahim |
| Hela | 0.0 -Hela | Isangi  | 0.0 -onomatopoea | 0.0 -Harens |
| Hoy na Gola | -0.296 -Either no | Kisangani | 0.0 -_ | 0.0 -Butticaz |
| Iboko | 0.0 -Trunk | Tomba | 0.0 -_ | 0.0 -Lieventhal |
| Ikeleso | 0.0 -Watch | _ | 0.0 -_ | 0.0 -_ |
| Ikoka | 0.0 -It is | _ | 0.0 -He who shoots people | 0.0 -_ |
| Ikuka | 0.0 -Shall we go | _ | -0.4215 -Bitter shrub | 0.0 -_ |
| Ilanga | 0.0 -Sun | Equateur | 0.0 -Enmity  | 0.0 -_ |
| Ipanga Ngunda | 0.0 -Will make the hind | Congo Free State | -0.5574 -He who destroys Upper Congo the country | 0.0 -_ |
| Ipipola | 0.0 -Pile | Opala | 0.0 -Copulation | 0.0 -_ |
| Itumba Mbilu | 0.0 -It is the heart | Tomba | 0.0 -War-like office | 0.0 -_ |
| Kabalo | 0.0 -know | Kasai | 0.0 -Horse | 0.0 -Gillian |
| Kabangu | 0.0 -KABANGU | Kasai | -0.5994 -Small scar; device of torture | 0.0 -L. De Brandt |
| Kabesa Babo | 0.0 -Kabesa babo | Lulubourg | 0.0 -_ | 0.0 -Wissman |
| Kabuakiatunge | 0.0 -Kability | Maniema | 0.0 -Whipping  | 0.0 -_ |
| Kakese | 0.0 -Kakes | Katanga | 0.0 -small, junior | 0.0 -_ |
| Kamuziki | 0.0 -Music | Lac Leopold II | 0.0 -Little music | 0.0 -A.J.G. Bolle |
| Kangipipe | 0.0 -Kangipipe | Katanga | 0.0 -_ | 0.0 -_ |
| Kapiteni | 0.0 -Captain | Province Orientale | 0.0 -Captain | 0.0 -generic |
| Kaputi | 0.0 -Coats | Inongo | 0.0 -_ | 0.0 -J. de Dixmude |
| Kaseya | 0.0 -Casea | Katanga | 0.0 -_ | 0.0 -_ |
| Kashabala | 0.0 -Glossy | Kasai | 0.0 -_ | 0.0 -Gasper |
| Kasiama Nkoi | 0.0 -Kasama Guy | _ | 0.0 -_ | 0.0 -_ |
| Kasongo Bushila | 0.0 -Kasongo Bushila | Kasai | 0.0 -Tall and lean | 0.0 -Bugschlag |
| Kasongo Mule | 0.0 -Kasongo Ie | Kasai | 0.0 -Tall and lean | 0.0 -Gillian |
| Kelekeke | 0.0 -Kelekeke | Katanga | 0.0 -_ | 0.0 -_ |
| Kelelo | 0.0 -Fish | Lac Leopold II | 0.0 -Bugle | 0.0 -A.J.G. Bolle |
| Kilupula | 0.0 -Kilupu | Katanga | 0.0 -_ | 0.0 -_ |
| Kimbwi | 0.0 -Kimfwi | Katanga | 0.0 -_ | 0.0 -_ |
| Kiomba Musinga | 0.0 -Kiroba you won't | Katanga | 0.0 -_ | 0.0 -_ |
| Kitatshindja | 0.0 -Kitty | Kasongo | -0.296 -He who cuts people in pieces | 0.0 -Moltedo |
| Kitoko | 0.0 -Wooden floor | everywhere | 0.8957 -Handsome; Beautiful; Delicious taste | 0.4939 -handsome agent |
| Kituanga | 0.0 -Repeat | Kisangi | 0.0 -He who smashes people; Grinder | 0.0 -Five Jean |
| Kobinda | 0.0 -Cop | _ | 0.0 -_ | 0.0 -Sergers |
| Koja | 0.0 -Which | Uele | 0.0 -_ | 0.0 -G. Junker |
| Kolongo | 0.0 -KONGO | Kisangani | 0.0 -_ | 0.0 -Lauwers Atay |
| Koma-Koma | 0.0 -Commercial | Equateur | 0.0 -He who can strangle me | 0.0 -_ |
| Komanda | 0.0 -A team | _ | 0.0 -Commander | 0.0 -generic |
| Konga | 0.0 -Mong | Katanga | 0.0 -_ | 0.0 -Crawford |
| Kulu-Kulu | 0.0 -Expense | Katanga | -0.7964 -Violence trigger; troublemaker | 0.0 -_ |
| Kuta Bongo | 0.0 -Brainstorms | Basoko Purnode | 0.0 -_ | 0.0 -_ |
| Lambo | 0.0 -Lambo | Katanga | 0.0 -_ | 0.0 -_ |
| Lamu | 0.0 -Lamu | Katanga | 0.0 -_ | 0.0 -_ |
| Liamamba | 0.0 -Heal | Ekonda, Opala | 0.0 -Millipede | 0.0 -_ |
| Liboma | 0.0 -Tie | Équateur | -0.5423 -He who kills | 0.0 -_ |
| Libumbu | 0.0 -Gym | Lefevre | 0.4767 -Elegance | 0.0 -_ |
| Libumu | 0.0 -lib u mother | Province Orientale | 0.0 -Belly | 0.0 -_ |
| Likoka | 0.0 -Lots | Tumba | 0.0 -Shooter | 0.0 -_ |
| Likoke | 0.0 -Letcha & Luke | Ekonda | -0.2263 -Prickly tree | 0.0 -_ |
| Likwama | 0.0 -Likwama | Mampoko | 0.0 -_ | 0.0 -M. Spelier |
| Lilanga-atumbe | 0.0 -The day is a Tumbi | Equateur | 0.0 -Garden destructor | 0.0 -_ |
| Limende | 0.0 -lime n | Yahisuli | 0.0 -_ | 0.0 -_ |
| Liombo | 0.0 -Limbo | Opala | 0.0 -Broom; Flywhisk | 0.0 -Wautier |
| Lipumbu | 0.0 -Prelimal | Province Orientale | 0.4767 -Elegant | 0.0 -_ |
| Lokesa | 0.0 -Lossa | Basoko | 0.0 -_ | 0.0 -Dekeyzer |
| Lokonga | 0.4939 -To save | Ekonda, Opala | 0.0 -venomous snake | 0.0 -_ |
| Lombolembo | 0.0 -Throat | Kisangani | 0.0 -_ | 0.0 -Jamsin |
| Longo-Longo | 0.0 -Long-erect | Kilo, Moto | 0.0 -tall-tall | 0.0 -_ |
| Longwango | 0.0 -LORE | Lac Leopold II | 0.0 -_ | 0.0 -G.O. Scioz |
| Longwani | -0.4767 -Fake | Katanga | 0.0 -_ | 0.0 -_ |
| Loponge | 0.0 -Loponge | Basoko | 0.0 -onomatopoea | 0.0 -P. Ponthier |
| Luanda | 0.0 -Luanda | Kazembe | 0.0 -Predatory trade | 0.0 -M. Caetano Pereira |
| Lubuku | 0.0 -Lie | Kasongo | 0.0 -Builder | 0.0 -Van Malderen |
| Lukwako | 0.0 -The giant | Zimba | 0.0 -Go back where you came from | 0.0 -_ |
| Lumandemulu | 0.0 -Lumandle | Lac Leopold II | 0.2023 -Fine collector | 0.0 -P.J. Van de Moere |
| Madami | 0.0 -a lot | Opala | 0.0 -Madam | 0.0 -generic |
| Mafuta | 0.0 -Oil | Katanga | 0.0 -fat | 0.0 -_ |
| Mafuta Mingi | 0.0 -Most fat | Mbesa, Beugamisa, Kilo | 0.0 -A lot of fat | 0.0 -Bell, M. Polis |
| Maine mingi | 0.0 -Reputation | Basoko | 0.0 -A lot of fat | 0.0 -_ |
| Maina | 0.0 -Maina | _ | 0.0 -Fat | 0.0 -_ |
| Makasi | -0.1027 -Hard | Aruwimi | 0.5106 -Force, power, strong  | 0.0 -_ |
| Makisi | 0.0 -Candals | Katanga | 0.0 -_ | 0.0 -_ |
| Makpatu | 0.0 -Makpatu | Basoko | 0.0 -Chaser of women | 0.0 -_ |
| Makupkup | 0.0 -Makupkup | Kuba | 0.0 -Of the old village | 0.0 -_ |
| Malakolo | 0.0 -Soft | Zimba | 0.0 -_ | 0.0 -_ |
| Mala-Mala | 0.0 -Mala-mala | Kasai | 0.0 -Main roads; Officials on the moove | 0.0 -_ |
| Malanda | 0.0 -Followers | Uele | 0.0 -Follower | 0.0 -Van Hende |
| Malenge | 0.0 -Pumpkin | Ubangi | 0.0 -_ | 0.0 -_ |
| Malonga Longa | 0.3612 -The tips like | Katanga | 0.0 -_ | 0.0 -_ |
| Malu-Malu | -0.25 -Shy | Lac Leopold II | 0.0 -Rush, rush | 0.0 -_ |
| Mandevu Mingi | 0.0 -Mindovu Mungi | Province Orientale | 0.0 -Beard | 0.0 -_ |
| Mangbe | 0.0 -Mangbe | Buta | 0.0 -Lightning; Scrapers of the heaps; White cannibals | 0.0 -_ |
| Mangema | 0.0 -Let | Zimba | 0.0 -_ | 0.0 -_ |
| Matala-Tala | 0.0 -Open-story | Buta | 0.0 -Mirror, eyeglasses, scrutiny; Witchcraft | 0.0 -_ |
| Matamba-Tamba | 0.0 -Cats-play | Opala | 0.0 -Enslavers | 0.0 -any agent |
| Matcho Kali | 0.0 -Match Cala | Opala | -0.5106 -Angry eyes; Harshness | 0.0 -_ |
| Matuba | 0.0 -Matuba | Territoire des cataractes | 0.0 -Words | 0.0 -_ |
| Mayala | 0.0 -Mayala | Territoire des cataractes | 0.0 -Oppressor | 0.0 -_ |
| Mbavu Munene | 0.0 -A thick rib | everywhere | 0.0 -Big chest | 0.0 -_ |
| Mbavu Nguvu | 0.5106 -Ribs strong | everywhere | 0.5106 -Strong chest | 0.0 -_ |
| Mfumu Mantese | 0.0 -King Manuse | nan | 0.0 -Prince who has things to say  | 0.0 -A. De Meulemeester |
| Miere-Miere | 0.0 -Honey-miece | Kazembe  | -0.1027 -Many knives; Ferocious and wild warrior | 0.0 -Mayo-Monteiro |
| Miso Minei | 0.0 -Miso Minei | Aruwimi | 0.0 -Leopard's eyes; Theif | 0.0 -_ |
| Miti | 0.0 -Trees | Kasongo | 0.0 -Trees | 0.0 -Middage |
| Moke | 0.0 -Moke | Aruwimi | 0.0 -Small, short | 0.0 -generic |
| Moke-Moke | 0.0 -Flour-moke | Uele-Itimbiri | 0.0 -Small, small; verry small | 0.0 -Verstracten |
| Molanda | 0.0 -Molanda | Uele | 0.0 -Follower | 0.0 -Hinde |
| Mondele Madami | 0.0 -Monother Madami | nan | 0.0 -Man who is always with is wife | 0.0 -Bandundu, Bulungu |
| Mondele Mboka | 0.0 -White country country | Aruwimi | 0.0 -Village white man; Native son; Head of the post | 0.0 -_ |
| Mondele na Kawa | 0.0 -The white man and kawa | Province Orientale | 0.0 -White man of coffee | 0.0 -generic |
| Mondele na Loso | 0.0 -The white man and the rice | Province Orientale | 0.0 -White man of rice | 0.0 -generic |
| Mondele na Mbila | 0.0 -The white man and the coconut | Province Orientale | 0.0 -White man of palm oil tree | 0.0 -generic |
| Mondele Ngolo | 0.0 -Meondo wagon | Bas Congo | 0.5106 -Strong white man | 0.0 -_ |
| Monganga na Mabele | 0.0 -doctor on the ground | Province Orientale | 0.0 -Earth healer; Mine prospector | 0.0 -_ |
| Monginda(Bonginda) | 0.0 -Monginda (Bonginda) | Tomba  | 0.0 -_ | 0.0 -Durieux |
| Mpimbo Mingi | 0.0 -There are so many whips | Equateur | 0.0 -Much whipping; hippopotamus-hide whip | 0.0 -_ |
| Msirkanda | 0.0 -Msirkanda | Province Orientale | 0.0 -_ | 0.0 -De Meulemeester |
| Mudiata ou Nyata | 0.0 -Mudiaata Ou is real | Katanga | 0.0 -_ | 0.0 -Thiry |
| Mukalenge Leka | 0.2732 -If you create paren | _ | 0.0 -Prince Leka | 0.0 -Legat |
| Mundele Kikufi | -0.6486 -Following the Dead | Territoire des cararactes | 0.0 -Short white man | 0.0 -_ |
| Mundele Mbisi | 0.0 -Too Lesson | Bas Congo | 0.0 -_ | 0.0 -_ |
| Mundele Ngolo | 0.0 -Wagon | Bas Congo | 0.5106 -Strong white man | 0.0 -_ |
| Mundele Nioka | 0.0 -Midwele naika | Bas Congo | 0.3612 -White man who is like a snake | 0.0 -_ |
| Mundele Nzazi | 0.0 -Follow me | Bas Congo | 0.0 -_ | 0.0 -_ |
| Munyololo | 0.0 -Munyololo | Eastern congo | 0.0 -Chain, rope | 0.0 -_ |
| Mupe | 0.0 -Give it | Eastern Congo | 0.0 -Priest | 0.0 -generic |
| Mupenda Batu | 0.0 -Off the Bunguage | Irumu, Ituri, Eastern Congo | 0.5719 -He who loves people | 0.0 -generic |
| Mupenda Kula | 0.0 -You can track a scan | Yanonge | 0.4215 -He who likes to eat | 0.0 -_ |
| Muzungu wa Pamba | 0.0 -Cotton white | Kongolo | 0.0 -White man of cotton | 0.0 -_ |
| Mwambe | 0.0 -Eight | Basoko | 0.0 -Eight; Eight lashes | 0.0 -_ |
| Mwana Mputu | 0.0 -Baby | Equateur | 0.0 -Child of Europe | 0.0 -H. Bombeeck  |
| Mwandami | 0.0 -MAN | _ | 0.0 -Madam | 0.0 -generic |
| Mwendo-Mwendo | 0.0 -Leg-leg | Kazembe | 0.0 -Travel; travel; Voyage at the foot of the world  | 0.0 -M. Caetano Pereira |
| Nabira | 0.0 -Accumulate | Uele | 0.0 -_ | 0.0 -G. Miani |
| Nafranki | 0.0 -Franki | Uele | 0.0 -He has white beard and long hair | 0.0 -G. Miani |
| Ndeke, Ndeke-Ndeke | 0.0 -Cleen, laugh-laugh | _ | 0.0 -Bird; Officials on the move | 0.0 -generic |
| Ndevu | 0.0 -Beard | Eastern Congo | 0.0 -Bearded white man | 0.0 -_ |
| Ndjole-Ndjole | 0.0 -Ndjole-Ndjole | Ekonda | 0.0 -Bearded white man | 0.0 -_ |
| Ndoki | -0.3612 -Witch | Kasongo | -0.3612 -Witch | 0.0 -Elter |
| Ngangabuka | 0.0 -I didn't look at | Kasai | 0.0 -Priest; God’s healer | 0.0 -_ |
| Nganga Nzambe | 0.0 -MANY ADACTION | Equateur, Province Orientale | 0.0 -Priest; God’s healer | 0.0 -_ |
| Ngolo Mingi | 0.0 -With those rounds | Bas Congo  | 0.0 -Much Power | 0.0 -_ |
| Ngona na batu | 0.4215 -Lol the rock | Equateur, Province Orientale | 0.0 -Nightly talking slit gong | 0.0 -_ |
| Ngongo Leteta | 0.0 -Ngongo Leteta | Eastern Congo | -0.5423 -Bad Official | 0.0 -generic |
| Nialakowomubu | 0.0 -Leaves | Lac Leopold II | 0.0 -_ | 0.0 -J.M.F. Tumers |
| Niangeniange | 0.0 -Ni Anniang, eh |  Kisangani  | 0.0 -_ | 0.0 -G. Geerearts |
| Nkake | 0.0 -Lightning | White Province Orientale, Équateur | 0.0 -Lightning; White cannibals   | 0.0 -_ |
| Nkangobeko | 0.0 -Hobby | Inongo | 0.0 -_ | 0.0 -Cloetens |
| Nkoso | 0.0 -The korom | _ | 0.0 -Parrot | 0.0 -_ |
| Nkoyi(Nkoi) | 0.0 -Nko Y doctor (Nukoy) | Lac Leopold II | 0.0 -Leopard | 0.0 -H. Dupuis Shaw |
| Nsoni Mingi | 0.0 -Very many | _ | -0.5413 -Very Shameful | 0.0 -J. Fievez |
| Ntange | 0.0 -Ntange | Lac Leopold II | 0.0 -Bed | 0.0 -_ |
| Nyimi-Nyimi | 0.0 -NEWS-SPECIAL | Kasai | -0.3182 -Red ant; lack of civilization | 0.0 -_ |
| Nyoka | 0.0 -Snake | Katanga | -0.4404 -Snake; Resentment | 0.0 -Wilson |
| Nzoku | 0.0 -Elephant | _ | 0.0 -Elephant | 0.0 -_ |
| Nzokumasi | 0.0 -Khosi | Bas Congo  | 0.0 -_ | 0.0 -_ |
| Ondele W'Ekonge | 0.0 -That is awaiting w'eyo | Opala | 0.0 -White men of urena lobata  | 0.0 -_ |
| Padiri | 0.0 -Padiri | Eastern Congo  | 0.0 -Priest | 0.0 -_ |
| Paipo | 0.0 -Paipo | Kilo, Moto | 0.0 -Pipe | 0.0 -_ |
| Pamba | 0.0 -Garnish | Lac Leopold II | 0.0 -_ | 0.0 -J. Fievez |
| Panzi | 0.0 -Grasshopper | Tomba | -0.5267 -Lying side | 0.0 -J. Fievez |
| Pebe | 0.0 -Pebe | Tomba | 0.0 -_ | 0.0 -Clark |
| Pete ya Mai ya Sombe | 0.0 -The Main | Province Orientale | 0.0 -Green stars; cassava leaves color | 0.0 -_ |
| Pole-Pole | 0.0 -Field-pole | Eastern Congo | 0.024 -Very calm; slow walker; lazy walker | 0.0 -generic |
| Rumaliza | 0.0 -RETURN | Eastern Congo | -0.4588 -Destroyer | 0.0 -_ |
| Sango | 0.0 -Sango | Eastern Congo, Equator | 0.0 -Father | 0.0 -_ |
| Sato | 0.0 -Sato | _ | 0.0 -_ | 0.0 -G. Casati |
| Sesa | 0.0 -than | Basoko | 0.0 -_ | 0.0 -Chaltin |
| Sikitele | 0.0 -sikitetele | Bandundu, Equator Province Orientale | 0.0 -Secretary | 0.0 -generic |
| Sikoti | 0.0 -Tail | _ | 0.0 -Much whipping; hippopotamus-hide whip | 0.0 -generic |
| Simba Bulaya | 0.0 -Lion Bulaya | Katanga, Eastern Congo | 0.0 -Europe's lion | 0.0 -_ |
| Singa | 0.0 -Element | Basoko | 0.0 -_ | 0.0 -Van Wert |
| Situka | 0.0 -We are stung | Kisangani | 0.0 -_ | 0.0 -Foerch |
| Sokele(Lokele) | 0.0 -Sole (Loke) | Lac Leopold II | 0.0 -_ | 0.0 -L. A. Borms |
| Soso Aleli | 0.0 -SOSO CRYS | Province Orientale | 0.0 -Rooster has crowed | 0.0 -_ |
| Soso Mombi | 0.0 -Soso Mombi | Chez Gobila | 0.0 -White feathered chicken | 0.0 -Janssen |
| Soso Mpembe | 0.0 -Mouse Mpombe | Lac Leopold II | 0.0 -White feathered chicken | 0.0 -Janssen |
| Sukuma | 0.0 -Imagine | Nyangwe | 0.0 -Push | 0.0 -Losange |
| Tange | 0.0 -Tange | Equateur | 0.0 -Bed | 0.0 -J. Fievez |
| Tata | 0.0 -Tata | Equateur | 0.0 -Father | 0.0 -J. Fievez |
| Tomansangu | 0.0 -Toman -san | Bas Congo  | 0.0 -Send news; News maker | 0.0 -_ |
| Tomilali | 0.0 -Tomila | Opala | 0.0 -Court | 0.0 -_ |
| Tomo | 0.0 -Tomo | Katanga | 0.0 -_ | 0.0 -_ |
| Toro | 0.0 -Visit | Basoko | 0.0 -_ | 0.0 -Lund |
| Tshienda Bitekete | 0.0 -Tshienda Bikekete | Kasai | -0.3612 -Lazy Walker | 0.0 -_ |
| Tshiwayawaya | 0.0 -T -wrinkles and Japanese | Kasai | 0.0 -Balance in breeze | 0.0 -_ |
| Tshoma-Tshoma | 0.0 -Tshoma-Tshoma | Aruwimi | 0.0 -He who burns people | 0.0 -C. Van de Laniotte |
| Tshombe Bululu | 0.0 -Shortly | Kasai | -0.4215 -Bitter cassava leaves | 0.0 -Gillian |
| Tuku-Tuku | 0.0 -Send-offers | Kongolo | 0.0 -Motorcycle | 0.0 -generic |
| Tuku-Tuku | 0.0 -Send-offers | Kisangani | 0.0 -Motorcycle | 0.0 -V. Rue |
| Tumbaku | 0.0 -Tobacco | Kisangani | 0.0 -Tobacco, sticky mouth and stained teeth | 0.0 -N. Tobback |
| Tumba Lombe | 0.0 -Touch the locust | Equateur | 0.0 -Home Burner | 0.0 -_ |
| Tumpa Mokono | 0.0 -Tump it | Moto | 0.4939 -Throw hand | 0.0 -_ |
| Vandebunduki | 0.0 -Vandebunduki | Lac Leopold II | 0.0 -Vande, the gunman | 0.0 -L. Van den Brooke |
| Vandenbuluki | 0.0 -Vandenbuluki | Lac Leopold II | -0.3612 -Vande, the witch | 0.0 -L. Van den Brooke |
| Vansina | 0.0 -Vansina | _ | 0.0 -Sit down | 0.0 -J. Vansina |
| Wai-Wai | 0.0 -Wai-Wai | Kamina | -0.4588 -Big troublemaker | 0.0 -generic |
| Yamba-Yamba | 0.0 -Shant | Kasongo | 0.0 -Defecate, defecate | 0.0 -Bosoni |


Now that we have performed the necessary evaluation of our names, lets save our data to a csv for easy of access.

In [256]:
data_2.columns = ['Local Name', 'Produced Translation', 'Location', 'Translation', 'Real Name']

data_2 = pd.DataFrame(rows)

data_2.to_csv(r'Sample of names & scores.csv', header=False, index=False)

This file has been saved to this repository.

## Conclusion

In this file we have used the [Vader](https://towardsdatascience.com/sentimental-analysis-using-vader-a3415fef7664) library to perform Sentiment Analysis on a list of roughly 270 names. Vader does not allow for the performance of multilingual Sentiment analysis. It was necessary to use another library [translate-api](https://pypi.org/project/translate-api/#getting-started)
What this means is that inputs from other languages are translated to english in order to calculate a sentiment score. I tried looking for a Natural Language Processing database in Swahili, Kikongo, & Lingala. But resources in computer science are sparse in these langauges. This certainly affects the output of this program, and is reflected in the csv file, where The official translations and their scores are shown alongside the translations and scores I produced.

For the reasons above, the words with translated sentiment should be weighted less in the consideration of the word's lasting impact today.